tests/runner/src/env.rs

```diff
@@ -520,6 +605,25 @@ impl Database for GreptimeDB {
     }
 }
 
+#[async_trait]
+impl Database for GreptimeDB {
+    async fn query(&self, ctx: QueryContext, query: String) -> Box<dyn Display> {
+        if ctx.context.contains_key("restart") && self.env.server_addrs.server_addr.is_none() {
+            self.env.restart_server(self).await;
```

Comment:
@J0HN50N133 note that the server restarts here. you will need to recreate the connections for mysql/postgres

if you want to keep it simple, you can create mysql/postgres connection for each `query`